chaincode/src/main/java/de/upb/cs/uc4/chaincode/CertificateChaincode.java

```diff
@@ -0,0 +1,124 @@
+package de.upb.cs.uc4.chaincode;
+
+import de.upb.cs.uc4.chaincode.model.*;
+import de.upb.cs.uc4.chaincode.util.CertificateContractUtil;
+import de.upb.cs.uc4.chaincode.util.GsonWrapper;
+import org.hyperledger.fabric.contract.Context;
+import org.hyperledger.fabric.contract.ContractInterface;
+import org.hyperledger.fabric.contract.annotation.Contract;
+import org.hyperledger.fabric.contract.annotation.Default;
+import org.hyperledger.fabric.contract.annotation.Transaction;
+import org.hyperledger.fabric.shim.ChaincodeStub;
+
+import java.util.ArrayList;
+
+@Contract(
+        name="UC4.Certificate"
+)
+@Default
+public class CertificateChaincode implements ContractInterface {
+
+    private final CertificateContractUtil cUtil = new CertificateContractUtil();
+
+    @Transaction()
+    public void initLedger(final Context ctx) {
+
+    }
+
+    /**
+     * Adds a certificate to the ledger.
+     * @param ctx transaction context providing access to ChaincodeStub etc.
+     * @param enrollmentId enrollmentId for the certificate to be added
+     * @param certificate certificate to be set for the given enrollmentId
+     * @return certificate on success, serialized error on failure
+     */
+    @Transaction()
+    public String addCertificate(final Context ctx, final String enrollmentId, final String certificate) {
+
+        ChaincodeStub stub = ctx.getStub();
+
+        ArrayList<InvalidParameter> invalidParams = new ArrayList<>();
+        invalidParams.addAll(getErrorForEnrollmentId(enrollmentId));
+        invalidParams.addAll(getErrorForCertificate(certificate));
```

Comment:
I totally get why you use those helper methods here, but what you are essentially doing is creating a list, to which two other list are added. This calls the ```new ArrayList()``` constructor three times in total, which seems wrong to me. Personally, i would consider just removing the helper methods, as they are very small, and go with
```
ArrayList<InvalidParameter> invalidParams = new ArrayList<>();
if(enrollmentId == null || enrollmentId.equals("")){
    invalidParams.add(cUtil.getEmptyEnrollmentIdParam());
}
if(certificate == null || certificate.equals("")){
    invalidParams.add([...]);
}
```
But that's just a personal preference 😄